sashimi-webapp/src/database/data-modifier/dataModifier.js

```diff
@@ -40,31 +50,68 @@ export default class dataModifier {
     if (typeof Promise === 'function') {
       return new Promise((resolve, reject) =>
         dataAdd.createNewFolder(organizationId, folderPath, folderId)
-        .then(data => resolve(data))
+        .then(folderObject => resolve(folderObject))
         .catch(sqlErr => reject(sqlErr))
       );
     } else {
       throw new exceptions.PromiseFunctionNotDefined();
     }
   }
 
-  static saveFile(fileId, file) {
+  static moveFile(fileId, newPath) {
     if (typeof Promise === 'function') {
       return new Promise((resolve, reject) =>
-        dataUpdate.saveFile(fileId, file)
-        .then(data => resolve(true))
-        .catch(sqlError => reject(sqlError))
+        sqlCommands.changeFilePath(fileId, newPath)
+        .then(() => resolve())
+        .catch(sqlErr => reject(sqlErr))
+      );
+    } else {
+      throw new exceptions.PromiseFunctionNotDefined();
+    }
+  }
+
+  static renameFileName(fileId, newFileName) {
+    if (typeof Promise === 'function') {
+      return new Promise((resolve, reject) =>
+        sqlCommands.changeFileName(fileId, newFileName)
+        .then(() => resolve())
+        .catch(sqlErr => reject(sqlErr))
       );
     } else {
       throw new exceptions.PromiseFunctionNotDefined();
     }
   }
 
+  static renameFolderName(folderId, newFolderName) {
+    if (typeof Promise === 'function') {
+      return new Promise((resolve, reject) =>
+        sqlCommands.changeFolderName(folderId, newFolderName)
+        .then(() => resolve())
+        .catch(sqlErr => reject(sqlErr))
+      );
+    } else {
+      throw new exceptions.PromiseFunctionNotDefined();
+    }
+  }
+
+  static saveFile(fileId, fileContent) {
+    if (typeof Promise === 'function') {
+      return new Promise((resolve, reject) => {
+        fileContent = resolveFileSaving(fileContent);
+        dataUpdate.saveFile(fileId, fileContent)
+        .then(() => resolve())
+        .catch(sqlError => reject(sqlError));
+      });
+    } else {
+      throw new exceptions.PromiseFunctionNotDefined();
+    }
+  }
```

Comment:
This code could be better written like this:

```javascript
static saveFile(fileId, fileContent) {
  fileContent = resolveFileSaving(fileContent);
  return dataUpdate.saveFile(fileId, fileContent)
}
```

The existing code is doing "superfluous wrapping".
You can read more about this Promise anti-pattern from [Promise-pattern-and-anti-pattern](http://www.datchley.name/promise-patterns-anti-patterns/),
or read my explanation here:

`dataUpdate.saveFile` is already returning a promise. 
- If there is any error coming from `dataUpdate.saveFile`,
  its error can be directly propagated to the function that is using this promise.
- And since, the error is not being handled here and got rejected immediately,  
  there is no need to create a new Promise just to reject and error.